### PR TITLE
Bugfix/loyalist reflection

### PR DESCRIPTION
**Pull request by Crotalus**
Extension PR of #2358

Missiles are redirected with a wider arc instead of insta-turning, should be working for aeon missiles as well

### CODE REVIEW

**Comment by PhilipJFryFAF** [comment]
very nice!

Tested and found some issues:
the missiles from the governor class cruiser (t2 uef) disappear after they get redirected (projectile lifetime is 30 seconds which should be enough to go back and forth)

missiles from strategic missile subs impact on the water surface way before the actual sub if the vessel is submerged (if the sub is on the water surface then the missile does hit)

**Comment by CookieNoob** [comment]
I think the army of the projectile should definetely be changed when it is reflected back. Allied tmd and shields wont work against the reflected projectiles (as they dont work now).

File: lua/defaultantiprojectile.lua [hunk 1]
```diff
@@ -134,13 +137,15 @@ MissileRedirect = Class(Entity) {
...
-                            proj:SetNewTargetGround(enemyPos)
-                        else
+                        elseif not enemy:BeenDestroyed() then
                             proj:SetNewTarget(enemy)
                             WaitSeconds(2)
-                            proj:SetNewTargetGround(enemy:GetPosition())
+                            enemyPos = enemy:GetPosition()
                         end
+
+                        enemyPos[2] = math.max(17.45, enemyPos[2]-0.05) -- aim at surface if submerged
```
[the remark this answers is not in the document]
Comment (Crotalus) [reply lua/defaultantiprojectile.lua]: Thx, haven't touched supcom lua in a while :) Fixed now

**Comment by Crotalus** [comment]
Tried to clone the projectile but for some reason it was destroyed immediately after it's created. It works fine if I manually choose a projectile bp but TIFCruiseMissile04 fails for instance. Maybe it needs to be created from a weapon muzzle? 

(tried <entity>:CreateProjectile(bpname, ...) )

**Comment by CookieNoob** [comment]
hm.. I never tried to create a projectile in thin air...
I'm not sure if `Entity:CreateProjectile(proj_bp, [ox, oy, oz], [dx, dy, dz] ` is the right lua command. There is no argument for the army that it is assigned to.

what exactly did you try and what was the error? I would assume that the syntax is something like
`local newMissile = oldMissile:CreateProjectile(oldMissile:GetBlueprint(), 0, 0, 0, 0, 0, 0)`
but that would probably create the same missile (and even worse) for the same army

`ChangeUnitArmy(unit,armyIndex)` probably doesnt work for projectiles, does it?

**Comment by aeoncleanse** [comment]
Merged as 733bd28f48e070a9880ea6ecc9ff8cd8b919b5ee - This is a great fix for now, even if it's not 100%

**Pull request closed without merging**